core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -619,6 +625,9 @@ private static String cannotStreamResultsForNonStreamingInputs(String inputs) {
     expr("coalesce('a','b')").ok();
     expr("coalesce('a','b','c')")
         .columnType("CHAR(1) NOT NULL");
+
+    sql("select COALESCE(mgr, 12) as m from EMP")
```

Comment:
I would also add a test for a query like this: `NULLIF(x, -1)`.
According to the doc [1] it is also converted to `CASE WHEN ...`

[1] https://calcite.apache.org/apidocs/org/apache/calcite/sql/fun/SqlCaseOperator.html

Reply:
IMO, we cannot deduce whether `NULLIF(x, -1)` is nullable or not when validating since we don't know whether `x` equals `1` or not.  Besides, there is already some test for it[1].

[1] https://github.com/apache/calcite/blob/master/core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java#L603